### PR TITLE
feat: added source_type in output  while listing connections list

### PR DESCRIPTION
**Pull request by SwatiT17**
/gcbrun

### CODE REVIEW

**Comment by SwatiT17** [comment]
Fixes issue: #792 data-validation connections list output is limited

File: data_validation/cli_tools.py [hunk 1]
```diff
@@ -988,9 +988,9 @@ def list_connections():
     """List all saved connections."""
     mgr = state_manager.StateManager()
     connections = mgr.list_connections()
-
     for conn_name in connections:
-        logging.info(f"Connection Name: {conn_name}")
+        source_type = mgr.get_connection_config(conn_name).get("source_type")
```
Comment (Raniksingh) [inline data_validation/cli_tools.py:992]: Should we also add target_type along with source? @nehanene15 any thoughts?

Comment (mahtodeepak05) [reply data_validation/cli_tools.py:992]: @Raniksingh 
AFAIU, connection listing implies databases whose connection details is added,  we can't confirm whether it is used as target when using DVT. source_type itself should suffice, it says to which source\database connection is pointing to.

**Review by nehanene15 — COMMENTED** [review]
Looks good! Maybe we can update this unit test to reflect the new output - https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/tests/unit/test_cli_tools.py#L117

**Review by Raniksingh — APPROVED** [review]

**Pull request merged**